### ISSUE TITLE
You shouldn't be sending telemetry to your own system without explicit consent

### ISSUE THREAD

**Issue by samuelcolvin** [traceloop/openllmetry#2420]
As per [`packages/traceloop-sdk/traceloop/sdk/telemetry.py`](https://github.com/traceloop/openllmetry/blob/5ca3d6fbb8f81cbc5e199743aa5a663f2993f1a4/packages/traceloop-sdk/traceloop/sdk/telemetry.py) you're currently sending usage information back to your own posthog (I assume), this is not stated in your readme or prominently in your docs.

This can be changed by setting the env var.

```bash
TRACELOOP_TELEMETRY=false
```

But without  any explicit warning, why would I know to do this?

The python ecosystem relied on everyone building packages follow a few best practices, and the whole system of open source would break down if every library you installed started phoning home with information about the environment in which it's used, and how.

We'd love to recommend openllmetry with Logfire, but really can't while this is enabled by default.

I strongly recommend you remove this "phone home" functionality.

**Comment by nirga** [comment]
Hey @samuelcolvin thanks for reporting this. Here's my elaborate answer:
* First of all, [this is explicitly stated in our docs under "Privacy"](https://www.traceloop.com/docs/openllmetry/privacy/telemetry), including all the data that we collect (and ofc - it's open source so you can see that). We don't identify you or track you by any means.
* **The main reason we added it** (I wrote the code! 😅 ) was to be able to see exceptions within the instrumentations. Since providers change their APIs quite often (which causes instrumentations to break), we wanted to know about them so we can address the issues quickly. 
* It's a common practice AFAIK - [Chroma does that](https://docs.trychroma.com/telemetry), and so does Posthog and others. It's always enabled by default and can be disabled. I think it's a common practice recently and as long as it's done in a transparent way (which I believe we do) - is totally acceptable.

Having said that - if you want to include our SDK in your product and disable the telemetry collection by default - how about we add a parameter in the init method and then you can explicitly disable it. WDYT?

Hope that helps clarify our intentions and rationale around telemetry collection.

**Comment by nirga** [comment]
>  If every one of those packages (or even 50%) sent telemetry as you do, the Python open source ecosystem would be a great deal more dangerous and problematic to use.

You're right. As I've also replied on twitter - this was a necessity for us to fix issues in the instrumentations fast. I think the instrumentation code tend to be highly coupled to the original SDKs and the fact that we're currently maintaining it (and not the original authors of those SDKs we wrap) means we need to work hard to keep up with changes we don't always even know about. This is what allowed us to get to a point where OpenLLMetry actually work for engineers that try it our 95% of the time. I think putting it in our SDK only (and not in the standalone instrumentations) and calling it out in the README file is a good compromise - agree?

**Comment by samuelcolvin** [comment]
Yup, that's definitely much better.

I would suggest you should be pretty explicit about what exception information you're collecting, e.g.:
* are you collecting the traceback? (which contains file paths which can contain sensitive information)
* are you scrubbing the exception message in any way?

Also, are you collecting IP address on the server side?

**Comment by nirga** [comment]
@samuelcolvin [no traceback, just the exception itself](https://github.com/traceloop/openllmetry/blob/5ca3d6fbb8f81cbc5e199743aa5a663f2993f1a4/packages/traceloop-sdk/traceloop/sdk/telemetry.py#L75). No IP address ofc. Again, the sole purpose is to allow us to fix exceptions easily.
